tests/core/test_optimizable.py

```diff
@@ -1276,5 +1276,90 @@ def test_load_save(self):
         self.assertAlmostEqual(adder1.J(), adder1_str_regen1.J())
 
 
+class OptClassSharedDOFs(Optimizable):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def J(self):
+        return sum(self.x)
+
+
+class TestOptimizableSharedDOFs(unittest.TestCase):
+    """
+    Test the DOFs sharing for the Optimizable classes
+    """
+
+    def test_adder_dofs_shared_change_vals(self):
+        adder_orig = OptClassSharedDOFs(x0=[1, 2, 3], names=["x", "y", "z"],
+                                        fixed=[False, False, True])
+        adder_shared_dofs = OptClassSharedDOFs(dofs=adder_orig._dofs)
```

Comment:
is accessing `adder_orig._dofs` directly recommended here? is the convention that variables starting with underscore are "internal"?

Reply:
I updated the code to remove the underscore. The DOFs object is made available because we are making use of it outside the parent Optimizable object.